### PR TITLE
Bug/softtypedeserialization

### PR DESCRIPTION
**Pull request by GearsAD**

### CODE REVIEW

**Comment by GearsAD** [comment]
Not ready for merging just yet, but teeing it up for tomorrow.

**Comment by GearsAD** [comment]
Must remove katabasis & anabasis logging before merging

File: src/DispatchPackedConversions.jl [hunk 1]
```diff
@@ -130,7 +135,24 @@ function convert(::Type{VariableNodeData}, d::PackedVariableNodeData)
...
+  catch ex
+      @error "Unable to deserialize soft type $(d.softtype)"
+      io = IOBuffer()
+      showerror(io, ex, catch_backtrace())
+      err = String(take!(io))
+      @error err
+  end
+  @info "Net result: $st"
 
   return VariableNodeData(M1,M2,M3,M4, d.BayesNetOutVertIDs,
```
Comment (dehann) [inline src/DispatchPackedConversions.jl]: we can move to the new compareAll functions as time goes on.

File: src/DispatchPackedConversions.jl [hunk 2]
```diff
@@ -198,14 +223,20 @@ function convert(
...
   usrfnc = convert(F, d.fnc)
-  # @show d.multihypo
   mhcat = parsemultihypostr(d.multihypo)
-  # TODO store threadmodel=MutliThreaded,SingleThreaded in persistence layer
+
+  # TODO -- improve prepgenericconvolution for hypotheses and certainhypo field recovery when deserializing
+  # reconstitute from stored data
   ccw = prepgenericconvolution(Graphs.ExVertex[], usrfnc, multihypo=mhcat)
-  return FunctionNodeData{CommonConvWrapper{typeof(usrfnc)}}(d.fncargvID, d.eliminated, d.potentialused, d.edgeIDs,
+  @warn "experimental certainhypo unpacking -- using override"
```
Comment (dehann) [inline src/DispatchPackedConversions.jl]: ask to also remove this printout please

Comment (GearsAD) [reply src/DispatchPackedConversions.jl]: Done!

**Comment by GearsAD** [comment]
Done and ready to merge - works!

**Comment by dehann** [comment]
I'm busy checking the issue here

**Pull request merged**